### DISCUSSION TITLE
read file

### DISCUSSION THREAD

**Discussion by inseongkim22** [community/community#27946] [Codespaces]
I am trying to read a file at Python. I am keep getting error messages says the file does not exist.
Can someone help me?

**Comment by airtower-luna** [comment]
Not without seeing the exact error message. :wink: 

If you're getting a `FileNotFoundError` most likely the file really doesn't exist, and you need to fix the path you give to `open()` or whatever function you use.

**Reply by inseongkim22** [reply]
Can you help me how to create the file? I was doing fine writing code and getting the right output on the terminal. So I thought I created the file.

**Reply by inseongkim22** [reply]
infile = open ("stockPrice.txt", "r")
print(infile.read())

**Reply by airtower-luna** [reply]
I can tell you how to create *a* file using Python. :snake: But if you want it to be *the file your program is trying to open* you'll need to provide a lot more details (and then it's still a "maybe"). :sweat_smile: 

In general, when you ask a programming question on a public forum (like here), it's best if you include at least:

* the code you're talking about (in your case: the Python code you're writing),
* the exact commands you're running (in your case: how you call your Python script, how you tried to prepare the file)
* the full output of those commands, especially error messages.

Minus any personal details, of course, and plus anything else that might be relevant. That way people can take a look, and hopefully someone will have an idea. :wink:

**Reply by airtower-luna** [reply]
Okay, so you're trying to open `stockPrice.txt` for reading. That means the file must exist in the directory *from which you're calling* the Python script. How to check which directory that is and which files exist there depends on the OS and shell you're using.

**Reply by inseongkim22** [reply]
I am using Github. I am a beginner. I created directory and file at the codespace.

**Reply by airtower-luna** [reply]
In that case you should have Linux tools available, and could use `ls -l` to see files in the current directory, or possibly `find .` to show a whole directory tree (but that might be *a lot*). Can you copy and paste from the shell you have for the codespace, or possibly show a screenshot? It's hard to guess from descriptions.
